bindings/matrix-sdk-ffi/src/room_preview.rs

```diff
@@ -1,9 +1,87 @@
-use matrix_sdk::{room_preview::RoomPreview as SdkRoomPreview, RoomState};
+use matrix_sdk::{
+    room_preview::{
+        JoinRoomPreviewAction as SdkJoinRoomPreviewAction, LeaveRoomPreviewAction as SdkLeaveRoomPreviewAction, RoomPreview as SdkRoomPreview,
+        RoomPreviewActions,
+    },
+    Client, RoomState,
+};
 use ruma::space::SpaceRoomJoinRule;
 
-/// The preview of a room, be it invited/joined/left, or not.
+use crate::{client::JoinRule, error::ClientError, room::Membership};
+
 #[derive(uniffi::Record)]
 pub struct RoomPreview {
+    pub info: RoomPreviewInfo,
```

Comment:
Why do we need the intermediary `RoomPreviewInfo` data structure?

Reply:
So we didn't need to move everything inside `RoomPreview` into a `Record` if we made it an `Object` (as is the case now since I'm removing the separate `actions` part).